pkg/codegen/importer/hcl2.go

```diff
@@ -0,0 +1,467 @@
+// Copyright 2016-2020, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package importer
+
+import (
+	"fmt"
+	"math"
+	"strings"
+
+	"github.com/pulumi/pulumi/pkg/v2/codegen/hcl2/model"
+	"github.com/pulumi/pulumi/pkg/v2/codegen/hcl2/syntax"
+	"github.com/pulumi/pulumi/pkg/v2/codegen/schema"
+	"github.com/pulumi/pulumi/pkg/v2/resource/deploy/providers"
+	"github.com/pulumi/pulumi/sdk/v2/go/common/resource"
+	"github.com/pulumi/pulumi/sdk/v2/go/common/util/contract"
+	"github.com/zclconf/go-cty/cty"
+)
+
+// Null represents Pulumi HCL2's `null` variable.
+var Null = &model.Variable{
+	Name:         "null",
+	VariableType: model.NoneType,
+}
+
+// GenerateHCL2Definition generates a Pulumi HCL2 definition for a given resource.
```

Comment:
Probably out of scope for this PR, but should we standardize on the PCL nomenclature here?

Reply:
We should standardize, yes, but I think that involves changing the package paths as well, so I'd like to take care of that in one fell swoop later on.